language core;

extend with #exceptions, #exception-type-declaration;

exception type = Nat

fn main(n : Nat) -> Nat {
    return (fn(x : Nat) {
        return throw(succ(0))
    }) (0)
}